lstchain/hdu/hdu_table.py

```diff
@@ -0,0 +1,464 @@
+        names=(
+            'OBS_ID', 'RA_PNT', 'DEC_PNT', 'ZEN_PNT', 'ALT_PNT', 'AZ_PNT', 'ONTIME', 'LIVETIME', 'DEADC', 'TSTART',
+            'TSTOP', 'OBJECT','N_TELS', 'TELLIST'),
+        dtype=('>i8', '>f4', '>f4', '>f4', '>f4', '>f4', '>f4', '>f4', '>f4', '>f8', '>f8', 'S20','>i8', 'S20'),
+        meta={'name': 'OBS_INDEX'}
+    )
+
+    obs_table = vstack(obs_table)
+    obs_table.meta['EXTNAME'] = 'OBS_INDEX'
+    obs_table.meta["HDUCLASS"] = "GADF"
+    obs_table.meta["HDUDOC"] = "https://github.com/open-gamma-ray-astro/gamma-astro-data-formats"
+    obs_table.meta["HDUVERS"] = "0.2"
+    obs_table.meta["HDUCLAS1"] = "INDEX"
+    obs_table.meta["HDUCLAS2"] = "OBS"
+    obs_table.meta["TELESCOP"] = "CTA"
+    obs_table.meta["INSTRUME"] = "LST-1"
+
+    obs_table.meta['MJDREFI'] = event_table.meta['MJDREFI']
+    obs_table.meta['MJDREFF'] = event_table.meta['MJDREFF']
+
+    filename_obs_table = 'obs-index.fits.gz'
+    obs = fits.BinTableHDU(obs_table)
+    hdu_list = fits.HDUList()
+    hdu_list.append(obs)
+    #fits.append(fits_dir+filename_obs_table, obs_table, header='OBS_INDEX')
+    hdu_list.writeto(str(fits_dir)+"/"+filename_obs_table, overwrite=True) #Make it to update
+
+    return
+
+def create_event_list(data, run_number, Source_name):
+
+    """Create the event_list HDUs from the given data and fill it up with the relevant values
+
+    Parameters
+    ----------
+        Data: DL2 data file
+                'astropy.table.QTable'
+        Run: Run number
+                Int
+        Source_name: Name of the source
+                Str
+    Returns
+    -------
+        Events HDU:  `astropy.io.fits.BinTableHDU`
+        GTI HDU:  `astropy.io.fits.BinTableHDU`
+        Pointing HDU:  `astropy.io.fits.BinTableHDU`
+    """
+    name=Source_name
+
+    #Initiate tables
+    event_table = Table()
+    gti_table = Table()
+    pointing_table = Table()
+
+    lam = 1000 #Average rate of triggered events, taken by hand for now
+    # Timing parameters
+    t_start = data['dragon_time'][0]
+    t_stop = data['dragon_time'][-1]
+    time = Time(data['dragon_time'], format='unix', scale="utc")
+
+    obs_time = t_stop-t_start
+
+    #Position parameters
+    focal = 28 * u.m
+    pos_x = data['reco_src_x'] * u .m
+    pos_y = data['reco_src_y'] * u .m
+    pointing_alt = data['alt_tel'] * u.rad
+    pointing_az = data['az_tel'] *u.rad
+
+    coord = camera_to_altaz(pos_x = pos_x, pos_y=pos_y, focal = focal,
+                    pointing_alt = pointing_alt, pointing_az = pointing_az,
+                    obstime = time)
+    coord_icrs = coord.icrs
+    coord_pointing = camera_to_altaz(pos_x = 0 * u.m, pos_y=0 * u.m, focal = focal,
+                pointing_alt = pointing_alt[0], pointing_az = pointing_az[0],
+                obstime = time[0])
+
+    ##########################################################################
+    ### Event table columns
+    event_table["EVENT_ID"] = Column(
```

Comment:
It's much easier and much less code to just use an astropy table than declaring these low-level fits objects.